libtransmission/torrent.cc

```diff
@@ -2453,6 +2453,37 @@ bool renameArgsAreValid(std::string_view oldpath, std::string_view newname)
         !tr_strvContains(newname, TR_PATH_DELIMITER);
 }
 
+bool renameNoCollision(tr_torrent const* tor, std::string_view oldpath, std::string_view newname)
```

Comment:
_(nit)_

- maybe a different function name to clarify this function makes no changes
- also, please avoid negative logic ("NoCollision") in function names because they're confusing.

Maybe inverti the return values and rename the function as `renameWouldCauseCollision()`?